src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageReadmeControl.xaml.cs

```diff
@@ -17,54 +19,123 @@ namespace NuGet.PackageManagement.UI
+#pragma warning disable CS0618 // Type or member is obsolete
+                var previewBuilder = new PreviewBuilder();
+#pragma warning restore CS0618 // Type or member is obsolete
+                previewBuilder.EditorOptions = options.GlobalOptions;
+                _markdownPreview = previewBuilder.Build();
+                descriptionMarkdownPreview.Content = _markdownPreview?.VisualElement;
+            }
+            return true;
+        }
+
         private void ReadmeViewModel_PropertyChanged(object sender, PropertyChangedEventArgs e)
         {
             if (e.PropertyName == nameof(ReadmePreviewViewModel.ReadmeMarkdown))
             {
-                NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(UpdateMarkdownAsync).PostOnFailure(nameof(PackageReadmeControl), nameof(ReadmeViewModel_PropertyChanged));
+                NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+                {
+                    if (!string.IsNullOrWhiteSpace(ReadmeViewModel.ReadmeMarkdown))
+                    {
+                        await UpdateMarkdownAsync(ReadmeViewModel.ReadmeMarkdown);
+                    }
+                }).PostOnFailure(nameof(PackageReadmeControl), nameof(ReadmeViewModel_PropertyChanged));
+            }
+            if (e.PropertyName == nameof(ReadmePreviewViewModel.IsBusy))
+            {
+                PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(IsBusy)));
+            }
+            if (e.PropertyName == nameof(ReadmePreviewViewModel.IsReadmeReady))
+            {
+                PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(IsReadmeReady)));
             }
         }
 
-        private async Task UpdateMarkdownAsync()
+        private async Task UpdateMarkdownAsync(string markdown)
         {
             try
             {
-                if (!string.IsNullOrWhiteSpace(ReadmeViewModel.ReadmeMarkdown))
+                UpdateBusy(true);
+                if (_markdownPreview is not null)
                 {
-                    await _markdownPreview.UpdateContentAsync(ReadmeViewModel.ReadmeMarkdown, ScrollHint.None);
+                    await TaskScheduler.Default;
+                    await _markdownPreview.UpdateContentAsync(markdown, ScrollHint.None);
                 }
             }
             catch (Exception ex) when (ex is ArgumentException || ex is InvalidOperationException)
             {
-                ReadmeViewModel.ErrorWithReadme = true;
-                ReadmeViewModel.ReadmeMarkdown = string.Empty;
-                await TelemetryUtility.PostFaultAsync(ex, nameof(ReadmePreviewViewModel));
+                if (_cancellationTokenSource.IsCancellationRequested)
+                {
+                    await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
This can resume after the UI has disappeared and crash VS, this needs to respect a cancellation token.